jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/editor/S3TreeNode.kt

```diff
@@ -77,68 +92,103 @@ class S3TreeDirectoryNode(private val bucket: S3VirtualBucket, parent: S3LazyLoa
             .contents()
             ?.filterNotNull()
             ?.filterNot { it.key() == key }
-            ?.map { S3TreeObjectNode(bucket, this, it.key(), it.size(), it.lastModified()) as S3TreeNode }
+            ?.map { S3TreeObjectNode(this, it.key(), it.size(), it.lastModified()) as S3TreeNode }
             ?: emptyList()
 
         return (folders + s3Objects).sortedBy { it.key } + continuation
     }
 }
 
-private val fileTypeRegistry = FileTypeRegistry.getInstance()
+interface S3Object {
+    val bucket: S3VirtualBucket
+    val key: String
+    val versionId: String?
+
+    val size: Long
+    val lastModified: Instant
 
-open class S3TreeObjectNode(val bucket: S3VirtualBucket, parent: S3LazyLoadParentNode?, key: String, val size: Long, val lastModified: Instant) :
-    S3LazyLoadParentNode(bucket.name, parent, key) {
+    fun fileName(): String
+}
 
+open class S3TreeObjectNode(parent: S3LazyLoadParentNode, key: String, override val size: Long, override val lastModified: Instant) :
+    S3LazyLoadParentNode(parent.bucket, parent, key), S3Object {
     var showHistory: Boolean = false
-    var responseIterator: Iterator<ListObjectVersionsResponse>? = null
-    private val fileType = fileTypeRegistry.getFileTypeByFileName(name)
 
     init {
-        fileType.takeIf { it !is UnknownFileType }?.icon.let { icon = it }
+        icon = FileTypeRegistry.getInstance().getFileTypeByFileName(key.substringAfterLast("/")).also { println("HELLO $it") }.icon
```

Comment:
delete or log instead

Reply:
It will be deleted, trying to triage windows failures

Reply:
Okay its removed now